### PR TITLE
[BUG] fix falsy random_state check in ConvTimeNetForecaster dataloader

### LINKED ISSUE #10052: [BUG] ConvTimeNetForecaster silently ignores random_state=0 due to falsy check
### Describe the bug

`ConvTimeNetForecaster.build_pytorch_train_dataloader` and its pred
dataloader use `if self.random_state:` to check whether to seed the
PyTorch Generator. This is a falsy check `random_state=0` evaluates
to `False`, so seed 0 is silently ignored and the generator runs without
a fixed seed.

This means results are not reproducible even though the user explicitly
passed `random_state=0`, with no error or warning raised.

Seed 0 is extremely common in research and tutorials.

### Location

`sktime/forecasting/convtimenet.py` lines 355 and 409

### Buggy code

```python
# line 355 and 409 - both dataloaders
if self.random_state:              # 0 is falsy - seed never set
    gen.manual_seed(self.random_state)
```

### Fix

```python
if self.random_state is not None:  # correctly handles seed 0
    gen.manual_seed(self.random_state)
```

### Note

The network class `networks/convtimenet/forecaster/_convtimenet.py`
and the classification base `classification/deep_learning/_pytorch.py`
already use the correct `is not None` pattern. This fix makes the
forecaster consistent with the rest of the codebase.

### Impact

Any user passing `random_state=0` gets non-reproducible results silently.

### PR DESCRIPTION
**Pull request by paramsureliya**
Fixes #10052

#### What does this implement/fix?

`ConvTimeNetForecaster.build_pytorch_train_dataloader` and its pred
dataloader both used `if self.random_state:` to decide whether to seed
the PyTorch Generator. This is a falsy check  `random_state=0`
evaluates to False, so seed 0 is silently ignored and results are not
reproducible even though the user explicitly passed `random_state=0`.

Fixed by changing both checks on lines 355 and 409 to
`if self.random_state is not None:`.

Note: the network class and classification base already use the correct
`is not None` pattern  this fix makes the forecaster consistent with
the rest of the codebase.

#### Does your contribution introduce a new dependency?

No.

#### What should a reviewer concentrate their feedback on?

- Lines 355 and 409 in `sktime/forecasting/convtimenet.py`
- Regression test in `sktime/forecasting/tests/test_convtimenet.py`

#### Did you add any tests?

Yes. Added `sktime/forecasting/tests/test_convtimenet.py` with a
regression test that patches `torch.Generator` and verifies
`manual_seed` is called with 0 when `random_state=0` is passed.

#### PR checklist
- [x] The PR title starts with [BUG]

### CODE REVIEW

**Review by fkiraly — DISMISSED** [review]
Thanks

**Comment by Tanuj-Taneja1** [comment]
Hey @paramsureliya, Great catch.
Hey @fkiraly, I think this issue wasnt caught as convtimenet ignores test_fit_idempotent .
I think the test had issue with torch based neural networks. By any chance has that been fixed yet.

Also do you think if there is any other way to check idempotency for all estimators which ignore test_fit_idempotent  .

**Comment by paramsureliya** [comment]
Hi, @fkiraly all tests are passing could you please review it.
